src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -11,113 +11,342 @@
+
+            _locationManager.RequestLocationUpdates(_reportInterval, (float)_movementThreshold, _locationCriteria, mojSluchacz, Looper.MainLooper);
+        }
+        
+        public uint ReportInterval
+        { 
+            get => _reportInterval;
+            set
+            {
+                _reportInterval = value;
+                RemoveUpdates();
+                RequestUpdates();
+            }
+        }
+
+        public double MovementThreshold
+        { 
+            get => _movementThreshold;
+            set
+            {
+                _movementThreshold = value;
+                RemoveUpdates();
+                RequestUpdates();
+            }
+        }
+
+        public uint? DesiredAccuracyInMeters
+        {
+            get => _desiredAccuracyInMeters;
+            set
+            {
+                _desiredAccuracyInMeters = value;
+                RemoveUpdates();
+                RequestUpdates();
+            }
+        }
+
+        #region MyListener
+
+        public class MyLocListen : Java.Lang.Object, ILocationListener
+        {
 
-		public Task<Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
-			=> GetGeopositionAsync();
+            public void OnLocationChanged(Location location)
+            {
+                DateTime date = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+                date = date.AddMilliseconds(location.Time);
+                if (date.AddMinutes(1) > DateTime.UtcNow)
+                { // only from last minute (we don't want to get some obsolete location)
+                    mbZmiany = true;
+                    moLocat = location;
+                }
+            }
 
-		public static async Task<GeolocationAccessStatus> RequestAccessAsync() => GeolocationAccessStatus.Allowed;
+            public void OnProviderDisabled(string provider)
+            {
+                throw new global::System.OperationCanceledException();
+            }
 
-		private LocationManager InitializeLocationProvider(double desiredAccuracy)
-		{
-			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+            public void OnProviderEnabled(string provider)
+            {
+                
+            }
 
-			var criteriaForLocationService = new Criteria
-			{
-				Accuracy = Accuracy.Coarse
-			};
+            public void OnStatusChanged(string provider, Availability status, Bundle extras)
+            {
+                // This method was deprecated in API level 29 (Android 10). This callback will never be invoked.
+            }
+        }
 
-			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
 
-			if (acceptableLocationProviders.Any())
-			{
-				_locationProvider = acceptableLocationProviders.First();
-			}
-			else
-			{
-				_locationProvider = String.Empty;
-			}
+        MyLocListen mojSluchacz = new MyLocListen();
```

Comment:
`mojSluchacz` ?

Reply:
polish version of myListener :)
Changing to english.